### DISCUSSION TITLE
NDK

### DISCUSSION THREAD

**Discussion by MatejMagat305** [termux/termux-app#3447] [Q&A] [answered]
hi, I was noticed the termux is installing ndk like c and c++ compiler, but it is not contains c header, my question is would it posible to add theese headers into termux system, or it is pactical imposible?
I ask before I has installed https://github.com/lzhiyong/termux-ndk and I have two clang now ...

**Comment by sylirre** [comment]
What headers are missing? Termux has a complete set of headers used for C and C++.

**Comment by MatejMagat305** [comment]
yes, but termux do not know build .apk for android and if I understand it is about missing headers for apk, it is not?

**Reply by sylirre** [reply, marked as answer]
NDK doesn't build APK file. It only processes `Android.mk` file and produces shared library or executables from C/C++ code. The headers available in Termux is a **copy** from official NDK nundle.

The APK file is built by Android SDK. Specifically by the `aapt` utility which normally invoked by build system.

APK does not require any headers. It is just a zip archive.
